### PR TITLE
Game feature

### PR DESCRIPTION
**Pull request by itsbaska**
Game is able to check if the board has 3 marks in a row. Given that the game has ended, the app will give the user a "win" or "tie" message.

### CODE REVIEW

File: test/messages_test.exs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                                  Computer Wins!!!
+                                  """
+  end
+
+  test "return game tie message" do
+    assert %Message{}.tie === """
+                                  It's a Tie!!!
+                                  """
+  end
+end
```
Comment (beccanelson) [inline test/messages_test.exs]: I would argue that this isn't a necessary test file. To me, `Message` is more of a config file than a file that contains actual game logic. In theory, we should be able to change config files without breaking any tests.

File: test/game_test.exs [hunk 2]
```diff
@@ -2,7 +2,76 @@ defmodule GameTest do
...
+  end
+
+  describe "when game is over" do
+    test "first row win" do
+      assert Game.game_over?(["O", "O", "O",
+                              "O", "X", "X",
+                              "X", "O", "O"]) == true
+    end
+
+    test "sencond row win" do
```
Comment (beccanelson) [inline test/game_test.exs]: ^ typo in "second"

File: test/game_test.exs [hunk 3]
```diff
@@ -2,7 +2,76 @@ defmodule GameTest do
...
+                              "X", "X", "X"]) == true
+    end
+
+    test "first column win" do
+      assert Game.game_over?(["X", "X", "O", 
+                              "X", "O", "X",
+                              "X", "O", "X"]) == true
+    end
+
+    test "sencond column win" do
```
Comment (beccanelson) [inline test/game_test.exs]: same typo here

File: test/game_test.exs [hunk 4]
```diff
@@ -2,7 +2,76 @@ defmodule GameTest do
...
+                              
+    end
+
+    test "tie" do
+      assert Game.game_over?(["X", "X", "O",
+                              "O", "O", "X",
+                              "X", "O", "X"]) == true
+    end
+  end
+
```
Comment (beccanelson) [inline test/game_test.exs]: Are you testing diagonal wins?

File: test/messages_test.exs (hunk 1 above)
Comment (esmevane) [reply test/messages_test.exs]: Agreed - there is a circumstance where you could reintroduce tests on files like this (Maybe for localization) - but as-is, it's probably OK to leave the tests out here.

**Pull request merged**